python/packages/ai/teams/ai/models/openai_model.py

```diff
@@ -271,7 +271,34 @@ async def complete_prompt(
                     if delta.content:
                         message_content += delta.content
 
-                    # TODO: Handle tool calls
+                    # Handle tool calls during streaming
+                    if is_tools_aug and delta.tool_calls:
+                        if not hasattr(message, "action_calls") or message.action_calls is None:
+                            message.action_calls = []
+
+                        for curr_tool_call in delta.tool_calls:
+                            index = curr_tool_call.index
+
+                            # Ensure the action_calls array is long enough
+                            while index >= len(message.action_calls):
```

Comment:
should this be in a while loop, or an if loop?

Reply:
ah i misunderstood the behavior, thinking I was monitoring its approach, not absolute status. My bad!